src/SRAssemblerMaster.cpp

```diff
@@ -1194,31 +1216,40 @@ void SRAssemblerMaster::remove_taboo_reads() {
 			// If there are fewer split read files than processors
 			if (lib.get_num_chunks() < mpiSize){
 				for (read_chunk=1; read_chunk<=lib.get_num_chunks(); read_chunk++){
+					logger->mpi("tabooing: sending read_chunk to prc" + int2str(read_chunk) + " in round " + int2str(round) + " for lib_idx " + int2str(lib_idx));
 					send_code(read_chunk, ACTION_ALIGNMENT, round, read_chunk, lib_idx);
 				}
 				while(completed < lib.get_num_chunks()){
 					mpi_receive(code_value, source);
+					completed++;
```

Comment:
You now have `completed ++` running twice per loop, because you did not delete it on the lower line.